index.ts

```diff
@@ -1,2 +1,46 @@
-import CashBalances from './data';
+import CashBalances, {CashBalanceReport, ReportSection, Account} from './data';
+
+interface ReportAccountTotals {
+  highest_account_name: string
+  lowest_account: string
+  average: any
+}
+
+/**
+ * Takes a report section and returns a flattened list of accounts
+ */
+function getFlattenedListOfAccounts(section: ReportSection): Account[] {
+  return [...section.accounts,
+    ...section.nestedSection[0].accounts, ...section.nestedSection[0].nestedSection[0].accounts,
+    ...section.nestedSection[0].nestedSection[0].nestedSection[0].accounts]
+}
+
+function getAverageAccountBalance(accounts: Account[]):number {
+  let balance = 0;
+
+  accounts.forEach(acc => {
+    balance += acc.value;
+  })
+
+  return balance / accounts.length;
+}
+
+function CalculateAccountTotals(accounts: Account[]): ReportAccountTotals {
```

Comment:
```suggestion
function CalculateAccountStats(accounts: Account[]): ReportAccountStats {
```
This might be pedantic but, none of these are totals.